check_api/src/main/java/com/google/errorprone/ErrorPronePlugins.java

```diff
@@ -45,11 +54,50 @@ public static ScannerSupplier loadPlugins(ScannerSupplier scannerSupplier, Conte
       return scannerSupplier;
     }
     ClassLoader loader = fileManager.getClassLoader(StandardLocation.ANNOTATION_PROCESSOR_PATH);
-    Iterable<BugChecker> extraBugCheckers = ServiceLoader.load(BugChecker.class, loader);
-    if (Iterables.isEmpty(extraBugCheckers)) {
+
+    List<Class<? extends BugChecker>> bugCheckerClasses;
+    // Synchronize to prevent races
+    synchronized(ErrorPronePlugins.class) {
+      // disable URL caching to avoid stale state bugs when run within a daemon-building process
```

Comment:
Given that it does have to be configured globally, and that it could be causing the same problem for javac's use of `ServiceLoader`, it still seems like disabling it in Buck would be a good idea. I suppose we could also disable it in Error Prone for insurance, though.

And thanks - Bazel does use a persistent javac worker that may be affected, I filed #3332 for that.